### ISSUE TITLE
[Blog list] Blog link generates overly verbose user experience

### ISSUE THREAD

**Issue by svinkle** [Shopify/dawn#692]
## Issue summary

The blog list item link wrap the entirety of the card content:

1. Image `alt` text
2. Post title
3. Post date and author
4. Post excerpt
5. …and other content

This generates an overly verbose user experience for screen reader and voice dictation users. Screen readers announce all content at once, or when navigating via virtual cursor, announce "link" along with each piece of content.

> "Welcome Home, Hem founder Petrus "Palmér" Welcome Home, Hem founder Petrus "Palmér" MARCH 2018SCOTT VINKLE Hem means home, and home is where the heart is. For Hem founder Petrus Palmér and his family, home is located in a remodeled blacksmith’s workshop situated in the Stockholm... &quot;Palmér&quot; 10 comments"

Voice dictation users may not be able to call out the command to click the link with the post title; this is not the first piece of content which makes up the computed accessible name (the first piece is the image `alt` as this is first in the DOM within the link.)

<details>
  <summary>Screenshots</summary>

  <img width="588" alt="Dawn blog list page, list item is highlighted." src="https://user-images.githubusercontent.com/1392632/135102692-e6188737-202a-4d85-9411-cd74b5c72ade.png">

</details>

#### HTML

```html
<a
  href="/blogs/news/welcome-home-hem-founder-petrus-palmer"
  class="article-content motion-reduce"
>
  <div class="article-card__image-wrapper">
    <div class="article-card__image media media--landscape">
      <!-- Other content… -->
    </div>
  </div>
  <div class="article-card__info">
    <header class="article-card__header">
      <h2 class="article-card__title" id="Article-10759274554">
        Welcome Home, Hem founder Petrus "Palmér"
      </h2>
      <!-- Other content… -->
    </header>
    <!-- Other content… -->
  </div>
</a>

<!-- Other blog list items… -->
```

## Steps to reproduce

1. Go to a Dawn theme store with any supported browser
2. Navigate to a blog list page
3. With an active screen reader, <kbd>Tab</kbd> to or use the virtual cursor to navigate the blog listing content

### Expected

- <kbd>Tab</kbd> to announce the post title only, virtual cursor to consume other list item content with intended semantic meaning.

### Actual

- <kbd>Tab</kbd> announces all content within the link, virtual cursor announces all content items with "link" semantics.

### HTML structure…

1. Move the link element to only wrap the post title text.

### CSS modifications

1. Use CSS to wrap/cover the card with the link element to re-apply the large click/touch target.

Review [Inclusive Components – Card](https://inclusive-components.design/cards/) for more the rationale behind these changes.

#### HTML

```html
<div class="article-card__image-wrapper">
  <div class="article-card__image media media--landscape">
    <!-- Other content… -->
  </div>
</div>
<div class="article-card__info">
  <header class="article-card__header">
    <h2 class="article-card__title" id="Article-10759274554">
      <a
        href="/blogs/news/welcome-home-hem-founder-petrus-palmer"
        class="article-content motion-reduce"
      >
        Welcome Home, Hem founder Petrus "Palmér"
      </a>
    </h2>
    <!-- Other content… -->
  </header>
  <!-- Other content… -->
</div>

<!-- Other blog list items… -->
```

#### CSS

```css
.article-card {
  position: relative;
  /* … */
}

a.article-content.motion-reduce:after {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
}
```

<!-- #### Diff

```diff
- line removed
+ line added
``` -->

## Specifications

- **Browser:** Any
- **Dawn version:** 2.1.0
- **Component:** Blog list
- **WCAG Principal:** Perceivable, Robust
- **WCAG SC:** [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html), [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html)
- **Severity:** High
- **Effort:** Medium
- **Related:** https://github.com/Shopify/dawn/issues/452

**Comment by kmeleta** [comment]
@svinkle is this still an issue? We've since updated article cards to share the same structure as product cards where the link is only around the title. This would also apply to #452

**Comment by svinkle** [comment]
@kmeleta Excellent. Just tried v4.0.0 and this looks great. I'll close this and #452.
